unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSComponentManagerLegacy/ECSComponentsManagerLegacy.cs

```diff
@@ -345,20 +345,13 @@ public ISharedComponent SceneSharedComponentCreate(string id, int classId)
         {
             if (disposableComponents.TryGetValue(id, out ISharedComponent component))
                 return component;
-
-            if (classId == (int)CLASS_ID.UI_SCREEN_SPACE_SHAPE || classId == (int)CLASS_ID.UI_FULLSCREEN_SHAPE)
+            
+            var factory = Environment.i.world.componentFactory;
```

Comment:
a reference to `Environment.i.world.componentFactory` already exist in this context (`this.componentFactory`)